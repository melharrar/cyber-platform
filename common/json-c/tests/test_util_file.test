#!/bin/sh

# Common definitions
if test -z "$srcdir"; then
    srcdir="${0%/*}"
    test "$srcdir" = "$0" && srcdir=.
    test -z "$srcdir" && srcdir=.
fi
. "$srcdir/test-defs.sh"

cp -f "$srcdir/valid.json" .
run_output_test test_util_file "$srcdir"
_err=$?

exit $_err
