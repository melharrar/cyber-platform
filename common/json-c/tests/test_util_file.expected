OK: json_object_from_fd(./valid.json)={ "foo": 123 }
OK: json_object_from_file(./not_present.json) correctly returned NULL: json_object_from_file: error opening file ./not_present.json: ERRNO=ENOENT

OK: json_object_from_fd(closed_fd), expecting NULL, EBADF, got:NULL, json_object_from_fd: error reading fd 10: ERRNO=EBADF

OK: json_object_to_file(json.out, jso)=0
file[json.out], size=336, contents={"foo":1234,"foo1":"abcdefghijklmnopqrstuvwxyz","foo2":"abcdefghijklmnopqrstuvwxyz","foo3":"abcdefghijklmnopqrstuvwxyz","foo4":"abcdefghijklmnopqrstuvwxyz","foo5":"abcdefghijklmnopqrstuvwxyz","foo6":"abcdefghijklmnopqrstuvwxyz","foo7":"abcdefghijklmnopqrstuvwxyz","foo8":"abcdefghijklmnopqrstuvwxyz","foo9":"abcdefghijklmnopqrstuvwxyz"}

OK: json_object_to_file_ext(json2.out, jso, JSON_C_TO_STRING_PRETTY)=0
file[json2.out], size=367, contents={
  "foo":1234,
  "foo1":"abcdefghijklmnopqrstuvwxyz",
  "foo2":"abcdefghijklmnopqrstuvwxyz",
  "foo3":"abcdefghijklmnopqrstuvwxyz",
  "foo4":"abcdefghijklmnopqrstuvwxyz",
  "foo5":"abcdefghijklmnopqrstuvwxyz",
  "foo6":"abcdefghijklmnopqrstuvwxyz",
  "foo7":"abcdefghijklmnopqrstuvwxyz",
  "foo8":"abcdefghijklmnopqrstuvwxyz",
  "foo9":"abcdefghijklmnopqrstuvwxyz"
}
